### ISSUE TITLE
Cannot pick folder location for downloads

### ISSUE THREAD

**Issue by SL4F** [rhunk/SnapEnhance#1107]
### Description

I cannot pick a location for downloads
![Screenshot_20240608-145529](https://github.com/rhunk/SnapEnhance/assets/52598005/ccadcfae-f446-4a65-b3fb-cc3b7fbdbb15)

I've tried on both stable and latest debug as of reporting.

### Reproduction steps

1. download and install 
2. start setting up
3. attempt to pick location 
4. cannot see or pick a location

### Snapchat Version

version 12.81.0.44 com.snapchat.android

### SnapEnhance Version

version 2.1.0 me.rhunk.snapenhance

### Agreement

- [X] This is not a bug regarding Snapchat+.
- [X] I have provided a detailed description of the issue.
- [X] I have attached a log if deemed necessary.
- [X] This issue is not a duplicate.

**Comment by SL4F** [comment]
this is my android version if it helps in any way:
VanillaIceCream (AP31.240426.023.B4)

**Comment by authorisation** [comment]
Looks like there is an issue with your files app

**Comment by SL4F** [comment]
> Looks like there is an issue with your files app

the app

**Comment by SL4F** [comment]
oops, the app works fine whenever opened by itself

**Comment by SL4F** [comment]
other apps also have no issues when using the files app.

**Comment by bocajthomas** [comment]
`Pick Folder` uses the `com.google.android.documentsui` file thing iirc. My best guess is that your device manufacturer has augmented it.

**Comment by authorisation** [comment]
> `Pick Folder` uses the `com.google.android.documentsui` file thing and not the files app it self iirc.

`com.google.android.documentsui` essentially acts as a proxy that handles backend related operations for the Google Files app.

> and I am pretty sure SnapEnhance doesn't provide support for android 15 yet since it's not currently supported because its a beta version.

SnapEnhance should support it fine, LSPosed however does not. I haven't tested this myself, as I am not in the possession of an Android 15 device.

**Comment by authorisation** [comment]
This may be related to A15, share a logcat.

**Comment by bocajthomas** [comment]
> SnapEnhance should support it fine, LSPosed however does not. I haven't tested this myself, as I am not in the possession of an Android 15 device.

@authorisation that's what I meant, 
https://github.com/rhunk/SnapEnhance/issues/1031#issuecomment-2111871228

**Comment by SL4F** [comment]
> @SL4F could you try [this build](https://github.com/rhunk/SnapEnhance/actions/runs/9431336772)? It should show more choices when selecting a folder

same issue, is it possible that it's because in the permissions it doesn't show storage as an option? I was never prompted to allow the app to allow files so it never got added, not really sure though.

![Screenshot_20240609-162954](https://github.com/rhunk/SnapEnhance/assets/52598005/fdb69f14-4ecd-4776-b3f6-b58c8ceb0c33)

**Comment by SL4F** [comment]
@rhunk this build let's me see and pick a folder but it keeps asking to pick a folder every time I open Snapchat and/or snapenhance. it also seems as if snapenhance is having troubles writing anything to storage, as exporting settings creates a file but seems to be corrupted.

https://github.com/rhunk/SnapEnhance/actions/runs/9456239072
![Screenshot_20240611-121027](https://github.com/rhunk/SnapEnhance/assets/52598005/b1ad1582-fdad-43f0-b19c-6736e2b1d0ef)
![Screenshot_20240611-121031](https://github.com/rhunk/SnapEnhance/assets/52598005/e3fdb5a2-1154-43d5-804d-0a7a74849e8c)
![Screenshot_20240611-121036](https://github.com/rhunk/SnapEnhance/assets/52598005/8c2ef5f8-0108-474f-99dc-6a4e0120a0de)

**Comment by SL4F** [comment]
> @SL4F have you tried creating a folder specifically for SE and using it in the picker?

yeah, I made a folder in downloads called Snapchat as seen in the screenshots.

**Comment by SL4F** [comment]
also, when installed with package manager on Android 15, it doesn't allow you to display over other apps or allow the phone permission unless the app comes from the play store, I have to install it via lucky patcher so it makes Android think it was downloaded from the play store, but I don't think that's the problem here.

**Comment by SL4F** [comment]
just checked with different folders and different locations and the same thing happens.

**Comment by bocajthomas** [comment]
@authorisation should slate be added to this too ? Last comment was made 1 week ago

**Comment by authorisation** [comment]
> @authorisation should slate be added to this too ? Last comment was made 1 week ago
> 
> Also refers to A15 too

No

**Comment by Keine-Ahnung-Bitch** [comment]
Same Problem

**Comment by SL4F** [comment]
@rhunk can you please fix this

https://github.com/rhunk/SnapEnhance/assets/52598005/97a22c10-bea9-466c-a293-0b5b4f64be1b

**Comment by D0lp1n** [comment]
I also had the same problem on my android 14 Superior extended os. I thought i had this issue cause i had this magisk module installed, "NoStorageRestrict" so i turned it off and my issue is gone. So maybe try remove magisk modules which interfere/interact with storage permissions or anything like that. Maybe it will fix the issue for you.

**Comment by aalqarnimr** [comment]
I had the same problem where files app keeps asking for folder location. When i whitelisted com.android.documentsui from Privacy Space app (some may try it on Hidemyapplist app) the problem disappeared
